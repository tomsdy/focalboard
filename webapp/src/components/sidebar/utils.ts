// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {CategoryBoards, DefaultCategory} from '../../store/sidebar'

import {Block} from '../../blocks/block'
import {Board} from '../../blocks/board'

export function addMissingItems(sidebarCategories: Array<CategoryBoards>, allItems: Array<Block | Board>): Array<CategoryBoards> {
    const blocksInCategories = new Map<string, boolean>()
    sidebarCategories.forEach(
        (category) => category.boardIDs.forEach(
            (boardID) => blocksInCategories.set(boardID, true),
        ),
    )

    const defaultCategory: CategoryBoards = {
        ...DefaultCategory,
        boardIDs: [],
    }

    allItems.forEach((block) => {
        if (!blocksInCategories.get(block.id)) {
            defaultCategory.boardIDs.push(block.id)
        }
    })

    // sidebarCategories comes from store,
    // so is frozen and can't be extended.
    // So creating new array from it.
    return [
        ...sidebarCategories,
        defaultCategory,
    ]
}
