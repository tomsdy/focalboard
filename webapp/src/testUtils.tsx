// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IntlProvider} from 'react-intl'
import React from 'react'

export const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

export function mockDOM(): void {
    window.focus = jest.fn()
    document.createRange = () => {
        const range = new Range()

        range.getBoundingClientRect = jest.fn()

        range.getClientRects = () => {
            return {
                item: () => null,
                length: 0,
                [Symbol.iterator]: jest.fn(),
            }
        }

        return range
    }
}
