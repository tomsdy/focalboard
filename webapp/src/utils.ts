// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import marked from 'marked'

declare global {
    interface Window {
        msCrypto: Crypto
    }
}

class Utils {
    static createGuid(): string {
        const crypto = window.crypto || window.msCrypto
        function randomDigit() {
            if (crypto && crypto.getRandomValues) {
                const rands = new Uint8Array(1)
                crypto.getRandomValues(rands)
                return (rands[0] % 16).toString(16)
            }

            return (Math.floor((Math.random() * 16))).toString(16)
        }
        return 'xxxxxxxx-xxxx-4xxx-8xxx-xxxxxxxxxxxx'.replace(/x/g, randomDigit)
    }

    static htmlToElement(html: string): HTMLElement {
        const template = document.createElement('template')
        template.innerHTML = html.trim()
        return template.content.firstChild as HTMLElement
    }

    static getElementById(elementId: string): HTMLElement {
        const element = document.getElementById(elementId)
        Utils.assert(element, `getElementById "${elementId}$`)
        return element!
    }

    static htmlEncode(text: string): string {
        return String(text).replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/"/g, '&quot;')
    }

    static htmlDecode(text: string): string {
        return String(text).replace(/&amp;/g, '&').replace(/&lt;/g, '<').replace(/&gt;/g, '>').replace(/&quot;/g, '"')
    }

    // Markdown

    static htmlFromMarkdown(text: string): string {
        // HACKHACK: Somehow, marked doesn't encode angle brackets
        const renderer = new marked.Renderer()
        renderer.link = (href, title, contents) => `<a target="_blank" rel="noreferrer" href="${href}" title="${title || ''}" onclick="event.stopPropagation();">${contents}</a>`
        const html = marked(text.replace(/</g, '&lt;'), {renderer, breaks: true})
        return html
    }

    // Date and Time

    static displayDate(date: Date): string {
        const dateTimeFormat = new Intl.DateTimeFormat('en', {year: 'numeric', month: 'short', day: '2-digit'})
        const text = dateTimeFormat.format(date)

        return text
    }

    static displayDateTime(date: Date): string {
        const dateTimeFormat = new Intl.DateTimeFormat(
            'en',
            {
                year: 'numeric',
                month: 'short',
                day: '2-digit',
                hour: 'numeric',
                minute: 'numeric',
            })
        const text = dateTimeFormat.format(date)
        return text
    }

    static sleep(miliseconds: number): Promise<void> {
        return new Promise((resolve) => setTimeout(resolve, miliseconds))
    }

    // Errors

    static assertValue(valueObject: any): void {
        const name = Object.keys(valueObject)[0]
        const value = valueObject[name]
        if (!value) {
            Utils.logError(`ASSERT VALUE [${name}]`)
        }
    }

    static assert(condition: any, tag = ''): void {
        /// #!if ENV !== "production"
        if (!condition) {
            Utils.logError(`ASSERT [${tag ?? new Error().stack}]`)
        }

        /// #!endif
    }

    static assertFailure(tag = ''): void {
        /// #!if ENV !== "production"
        Utils.assert(false, tag)

        /// #!endif
    }

    static log(message: string): void {
        /// #!if ENV !== "production"
        const timestamp = (Date.now() / 1000).toFixed(2)
        // eslint-disable-next-line no-console
        console.log(`[${timestamp}] ${message}`)

        /// #!endif
    }

    static logError(message: string): void {
        /// #!if ENV !== "production"
        const timestamp = (Date.now() / 1000).toFixed(2)
        // eslint-disable-next-line no-console
        console.error(`[${timestamp}] ${message}`)

        /// #!endif
    }

    // favicon

    static setFavicon(icon?: string): void {
        const href = icon ? `data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100"><text y=".9em" font-size="90">${icon}</text></svg>` : ''
        const link = (document.querySelector("link[rel*='icon']") || document.createElement('link')) as HTMLLinkElement
        link.type = 'image/x-icon'
        link.rel = 'shortcut icon'
        link.href = href
        document.getElementsByTagName('head')[0].appendChild(link)
    }

    // URL

    static replaceUrlQueryParam(paramName: string, value?: string): void {
        const queryString = new URLSearchParams(window.location.search)
        const currentValue = queryString.get(paramName) || ''
        if (currentValue !== value) {
            const newUrl = new URL(window.location.toString())
            if (value) {
                newUrl.searchParams.set(paramName, value)
            } else {
                newUrl.searchParams.delete(paramName)
            }
            window.history.pushState({}, document.title, newUrl.toString())
        }
    }

    // File names

    static sanitizeFilename(filename: string): string {
        // TODO: Use an industrial-strength sanitizer
        let sanitizedFilename = filename
        const illegalCharacters = ['\\', '/', '?', ':', '<', '>', '*', '|', '"', '.']
        illegalCharacters.forEach((character) => {
            sanitizedFilename = sanitizedFilename.replace(character, '')
        })
        return sanitizedFilename
    }

    // File picker

    static selectLocalFile(onSelect?: (file: File) => void, accept = '.jpg,.jpeg,.png'): void {
        const input = document.createElement('input')
        input.type = 'file'
        input.accept = accept
        input.onchange = async () => {
            const file = input.files![0]
            onSelect?.(file)
        }

        input.style.display = 'none'
        document.body.appendChild(input)
        input.click()

        // TODO: Remove or reuse input
    }

    // Arrays

    static arraysEqual(a: readonly any[], b: readonly any[]): boolean {
        if (a === b) {
            return true
        }
        if (a === null || b === null) {
            return false
        }
        if (a === undefined || b === undefined) {
            return false
        }
        if (a.length !== b.length) {
            return false
        }

        for (let i = 0; i < a.length; ++i) {
            if (a[i] !== b[i]) {
                return false
            }
        }
        return true
    }

    static arrayMove(arr: any[], srcIndex: number, destIndex: number): void {
        arr.splice(destIndex, 0, arr.splice(srcIndex, 1)[0])
    }

    // Clipboard

    static copyTextToClipboard(text: string): boolean {
        const textField = document.createElement('textarea')
        textField.innerText = text
        textField.style.position = 'fixed'
        textField.style.opacity = '0'

        document.body.appendChild(textField)
        textField.select()

        let result = false
        try {
            result = document.execCommand('copy')
        } catch (err) {
            Utils.logError(`copyTextToClipboard ERROR: ${err}`)
            result = false
        }
        textField.remove()

        return result
    }

    static isMobile() {
        const toMatch = [
            /Android/i,
            /webOS/i,
            /iPhone/i,
            /iPad/i,
            /iPod/i,
            /BlackBerry/i,
            /Windows Phone/i,
        ]

        return toMatch.some((toMatchItem) => {
            return navigator.userAgent.match(toMatchItem)
        })
    }
}

export {Utils}
