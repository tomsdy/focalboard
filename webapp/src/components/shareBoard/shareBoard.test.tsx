// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {act, render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import React from 'react'
import {MemoryRouter} from 'react-router'
import {mocked} from 'ts-jest/utils'

import {ISharing} from '../../blocks/sharing'
import {TestBlockFactory} from '../../test/testBlockFactory'
import {wrapDNDIntl} from '../../testUtils'
import client from '../../octoClient'
import {Utils} from '../../utils'

import ShareBoard from './shareBoard'

jest.useFakeTimers()

const boardId = '1'
const workspaceId: string|undefined = boardId
const viewId = boardId

jest.mock('../../octoClient')
jest.mock('../../utils')

const mockedOctoClient = mocked(client, true)
const mockedUtils = mocked(Utils, true)

let params = {}
jest.mock('react-router', () => {
    const originalModule = jest.requireActual('react-router')

    return {
        ...originalModule,
        useRouteMatch: jest.fn(() => {
            return {
                url: 'http://localhost/',
                path: '/',
                params,
                isExact: true,
            }
        }),
    }
})
const board = TestBlockFactory.createBoard()
board.id = boardId

describe('src/components/shareBoard/shareBoard', () => {
    const w = (window as any)
    const oldBaseURL = w.baseURL

    beforeEach(() => {
        jest.clearAllMocks()
        mockedUtils.buildURL.mockImplementation((path) => (w.baseURL || '') + path)

        params = {
            boardId,
            viewId,
            workspaceId,
        }
    })

    afterEach(() => {
        w.baseURL = oldBaseURL
    })

    test('should match snapshot', async () => {
        mockedOctoClient.getSharing.mockResolvedValue(undefined)
        let container
        await act(async () => {
            const result = render(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />),
                {wrapper: MemoryRouter},
            )
            container = result.container
        })

        expect(container).toMatchSnapshot()
        const closeButton = screen.getByRole('button', {name: 'Close dialog'})
        expect(closeButton).toBeDefined()
    })

    test('should match snapshot with sharing', async () => {
        const sharing:ISharing = {
            id: boardId,
            enabled: true,
            token: 'oneToken',
        }
        mockedOctoClient.getSharing.mockResolvedValue(sharing)

        let container
        await act(async () => {
            const result = render(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />),
                {wrapper: MemoryRouter},
            )
            container = result.container
        })
        const copyLinkElement = screen.getByRole('button', {name: 'Copy internal link'})
        expect(copyLinkElement).toBeDefined()

        expect(container).toMatchSnapshot()
    })

    test('return shareBoard and click Copy link', async () => {
        const sharing:ISharing = {
            id: boardId,
            enabled: true,
            token: 'oneToken',
        }
        mockedOctoClient.getSharing.mockResolvedValue(sharing)

        let container
        await act(async () => {
            const result = render(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />),
                {wrapper: MemoryRouter},
            )
            container = result.container
        })

        expect(container).toMatchSnapshot()

        const copyLinkElement = screen.getByRole('button', {name: 'Copy internal link'})
        expect(copyLinkElement).toBeDefined()

        await act(async () => {
            userEvent.click(copyLinkElement!)
        })

        expect(mockedUtils.copyTextToClipboard).toBeCalledTimes(1)
        expect(container).toMatchSnapshot()

        const copiedLinkElement = screen.getByRole('button', {name: 'Copy internal link'})
        expect(copiedLinkElement).toBeDefined()
        expect(copiedLinkElement.textContent).toContain('Copied!')
    })

    test('return shareBoard and click Regenerate token', async () => {
        window.confirm = jest.fn(() => {
            return true
        })
        const sharing:ISharing = {
            id: boardId,
            enabled: true,
            token: 'oneToken',
        }
        mockedOctoClient.getSharing.mockResolvedValue(sharing)

        let container
        await act(async () => {
            const result = render(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />),
                {wrapper: MemoryRouter},
            )
            container = result.container
        })

        sharing.token = 'anotherToken'
        mockedUtils.createGuid.mockReturnValue('anotherToken')
        mockedOctoClient.getSharing.mockResolvedValue(sharing)
        mockedOctoClient.setSharing.mockResolvedValue(true)

        const publishButton = screen.getByRole('button', {name: 'Publish'})
        expect(publishButton).toBeDefined()
        userEvent.click(publishButton)
        await act(async () => {
            jest.runOnlyPendingTimers()
        })

        const regenerateTokenElement = screen.getByRole('button', {name: 'Regenerate token'})
        expect(regenerateTokenElement).toBeDefined()
        userEvent.click(regenerateTokenElement)
        await act(async () => {
            jest.runOnlyPendingTimers()
        })
        expect(mockedOctoClient.setSharing).toBeCalledTimes(1)
        expect(container).toMatchSnapshot()
    })
    test('return shareBoard, and click switch', async () => {
        const sharing:ISharing = {
            id: boardId,
            enabled: true,
            token: 'oneToken',
        }
        mockedOctoClient.getSharing.mockResolvedValue(sharing)
        let container: Element | undefined
        await act(async () => {
            const result = render(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />),
                {wrapper: MemoryRouter},
            )
            container = result.container
        })

        const publishButton = screen.getByRole('button', {name: 'Publish'})
        expect(publishButton).toBeDefined()
        userEvent.click(publishButton)
        await act(async () => {
            jest.runOnlyPendingTimers()
        })

        const switchElement = container?.querySelector('.Switch')
        expect(switchElement).toBeDefined()
        await act(async () => {
            userEvent.click(switchElement!)
        })

        expect(mockedOctoClient.setSharing).toBeCalledTimes(1)
        expect(mockedOctoClient.getSharing).toBeCalledTimes(2)
        expect(container).toMatchSnapshot()
    })
    test('return shareBoardComponent and click Switch without sharing', async () => {
        mockedOctoClient.getSharing.mockResolvedValue(undefined)
        mockedUtils.createGuid.mockReturnValue('aToken')
        let container: Element | undefined
        await act(async () => {
            const result = render(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />),
                {wrapper: MemoryRouter},
            )
            container = result.container
            mockedOctoClient.getSharing.mockResolvedValue({
                id: boardId,
                enabled: true,
                token: 'aToken',
            })

            const publishButton = screen.getByRole('button', {name: 'Publish'})
            expect(publishButton).toBeDefined()
            userEvent.click(publishButton)
            jest.runOnlyPendingTimers()

            const switchElement = container?.querySelector('.Switch')
            expect(switchElement).toBeDefined()
            userEvent.click(switchElement!)
            jest.runOnlyPendingTimers()
            result.rerender(
                wrapDNDIntl(
                    <ShareBoard
                        boardId={board.id}
                        onClose={jest.fn()}
                        enableSharedBoards={true}
                    />))
        })

        expect(mockedOctoClient.setSharing).toBeCalledTimes(1)
        expect(mockedOctoClient.getSharing).toBeCalledTimes(2)
        expect(mockedUtils.createGuid).toBeCalledTimes(1)
        expect(container).toMatchSnapshot()
    })
    test('should match snapshot with sharing and without workspaceId and subpath', async () => {
        w.baseURL = '/test-subpath/plugins/boards'
        const sharing:ISharing = {
            id: boardId,
            enabled: true,
            token: 'oneToken',
        }
        params = {
            boardId,
            viewId,
        }
        mockedOctoClient.getSharing.mockResolvedValue(sharing)
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ShareBoard
                    boardId={board.id}
                    onClose={jest.fn()}
                    enableSharedBoards={true}
                />), {wrapper: MemoryRouter})
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot with sharing and subpath', async () => {
        w.baseURL = '/test-subpath/plugins/boards'
        const sharing:ISharing = {
            id: boardId,
            enabled: true,
            token: 'oneToken',
        }
        mockedOctoClient.getSharing.mockResolvedValue(sharing)
        let container
        await act(async () => {
            const result = render(wrapDNDIntl(
                <ShareBoard
                    boardId={board.id}
                    onClose={jest.fn()}
                    enableSharedBoards={true}
                />), {wrapper: MemoryRouter})
            container = result.container
        })
        expect(container).toMatchSnapshot()
    })
})
