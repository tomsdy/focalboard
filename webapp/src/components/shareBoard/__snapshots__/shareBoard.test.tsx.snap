// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/shareBoard/shareBoard return shareBoard and click Copy link 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item tab-item--active"
          >
            Share
          </button>
          <button
            class="tab-item false"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Share internally
                </div>
                <div
                  class="text-light"
                >
                  Users who have permissions will be able to use this link
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/undefined/workspace/1/1/1"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/undefined/workspace/1/1/1
              </a>
            </div>
            <button
              title="Copy internal link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard return shareBoard and click Copy link 2`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item tab-item--active"
          >
            Share
          </button>
          <button
            class="tab-item false"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Share internally
                </div>
                <div
                  class="text-light"
                >
                  Users who have permissions will be able to use this link
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/undefined/workspace/1/1/1"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/undefined/workspace/1/1/1
              </a>
            </div>
            <button
              title="Copy internal link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copied!
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard return shareBoard and click Regenerate token 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item false"
          >
            Share
          </button>
          <button
            class="tab-item tab-item--active"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Publish to the web
                </div>
                <div
                  class="text-light"
                >
                  Publish and share a “read only” link with everyone on the web
                </div>
              </div>
              <div>
                <div
                  class="Switch override-main size--medium on"
                >
                  <div
                    class="octo-switch-inner"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/workspace/1/shared/1/1?r=anotherToken"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/workspace/1/shared/1/1?r=anotherToken
              </a>
              <div
                class="octo-tooltip tooltip-top"
                data-tooltip="Regenerate token"
              >
                <button
                  aria-label="Regenerate token"
                  title="Regenerate token"
                  type="button"
                >
                  <i
                    class="CompassIcon icon-refresh undefined"
                  />
                </button>
              </div>
            </div>
            <button
              title="Copy public link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard return shareBoard, and click switch 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item false"
          >
            Share
          </button>
          <button
            class="tab-item tab-item--active"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Publish to the web
                </div>
                <div
                  class="text-light"
                >
                  Publish and share a “read only” link with everyone on the web
                </div>
              </div>
              <div>
                <div
                  class="Switch override-main size--medium on"
                >
                  <div
                    class="octo-switch-inner"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/workspace/1/shared/1/1?r=oneToken"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/workspace/1/shared/1/1?r=oneToken
              </a>
              <div
                class="octo-tooltip tooltip-top"
                data-tooltip="Regenerate token"
              >
                <button
                  aria-label="Regenerate token"
                  title="Regenerate token"
                  type="button"
                >
                  <i
                    class="CompassIcon icon-refresh undefined"
                  />
                </button>
              </div>
            </div>
            <button
              title="Copy public link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard return shareBoardComponent and click Switch without sharing 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item false"
          >
            Share
          </button>
          <button
            class="tab-item tab-item--active"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Publish to the web
                </div>
                <div
                  class="text-light"
                >
                  Publish and share a “read only” link with everyone on the web
                </div>
              </div>
              <div>
                <div
                  class="Switch override-main size--medium on"
                >
                  <div
                    class="octo-switch-inner"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/workspace/1/shared/1/1?r=aToken"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/workspace/1/shared/1/1?r=aToken
              </a>
              <div
                class="octo-tooltip tooltip-top"
                data-tooltip="Regenerate token"
              >
                <button
                  aria-label="Regenerate token"
                  title="Regenerate token"
                  type="button"
                >
                  <i
                    class="CompassIcon icon-refresh undefined"
                  />
                </button>
              </div>
            </div>
            <button
              title="Copy public link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard should match snapshot 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item tab-item--active"
          >
            Share
          </button>
          <button
            class="tab-item false"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Share internally
                </div>
                <div
                  class="text-light"
                >
                  Users who have permissions will be able to use this link
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/undefined/workspace/1/1/1"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/undefined/workspace/1/1/1
              </a>
            </div>
            <button
              title="Copy internal link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard should match snapshot with sharing 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item tab-item--active"
          >
            Share
          </button>
          <button
            class="tab-item false"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Share internally
                </div>
                <div
                  class="text-light"
                >
                  Users who have permissions will be able to use this link
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/undefined/workspace/1/1/1"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/undefined/workspace/1/1/1
              </a>
            </div>
            <button
              title="Copy internal link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard should match snapshot with sharing and subpath 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item tab-item--active"
          >
            Share
          </button>
          <button
            class="tab-item false"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Share internally
                </div>
                <div
                  class="text-light"
                >
                  Users who have permissions will be able to use this link
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/undefined/workspace/1/1/1"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/undefined/workspace/1/1/1
              </a>
            </div>
            <button
              title="Copy internal link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/components/shareBoard/shareBoard should match snapshot with sharing and without workspaceId and subpath 1`] = `
<div>
  <div
    class="Dialog dialog-back ShareBoardDialog"
  >
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <h1
            class="text-heading5 mt-2"
          >
             
          </h1>
          <button
            aria-label="Close dialog"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
        </div>
        <div
          class="tabs-container"
        >
          <button
            class="tab-item tab-item--active"
          >
            Share
          </button>
          <button
            class="tab-item false"
          >
            Publish
          </button>
        </div>
        <div
          class="tabs-content"
        >
          <div>
            <div
              class="d-flex justify-content-between"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="text-heading2"
                >
                  Share internally
                </div>
                <div
                  class="text-light"
                >
                  Users who have permissions will be able to use this link
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex justify-content-between tabs-inputs"
          >
            <div
              class="d-flex input-container"
            >
              <a
                class="shareUrl"
                href="http://localhost/test-subpath/plugins/boards1/1"
                rel="noreferrer"
                target="_blank"
              >
                http://localhost/test-subpath/plugins/boards1/1
              </a>
            </div>
            <button
              title="Copy internal link"
              type="button"
            >
              <i
                class="CompassIcon icon-content-copy CompassIcon"
              />
              <span>
                Copy link
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
